plutus-core/plutus-ir/test/prettyprintingReadable/even3.golden

```diff
@@ -0,0 +1,14 @@
+letrec data Nat | match_Nat where
+         Zero : Nat
+         Suc : Nat -> Nat
+in let data Bool | match_Bool where
+         True : Bool
+         False : Bool
+       !three : Nat = Suc (Suc (Suc Zero))
+   in letrec !even : Nat -> Bool
```

Comment:
I think not indenting subsequent lets is a good idea, which seems to happen on line 4 but not here for some reason?

Reply:
I still care about this comment! I hadn't quite processed it, but I see that it's related to the previous one: this is consistently aligned with the preceding `let`, so if we did
```
let
in
let
in
```
then we could have the same rule consistently and also get nonincreasing indentation.

I think it is _quite_ desirable to get nonincreasing indendation for nested lets, because PIR programs of any reasonable size tend to have a *lot* of them, is there a way we can do that nicely? If we have to do the unorthodox indentation of the lets then I can live with that.

Reply:
I buy having non-increasing indentation being a nice thing. The thing to ensure is that non-let terms below a let should still be indented, otherwise programs just look _weird_. I'll implement this option.

Reply:
The unorthodox one does avoid this by having the body-terms *always* be at the same indentation level. But perhaps it's too weird.

Reply:
Also I'm not sure I see the point about line 4 here. On that line we also indent?

Reply:
(Which we would under the orthodox scheme and wouldn't under the unorthodox scheme)

Reply:
Yep, `in let` and `letrec` have the same length. Ha.

Reply:
Ok, I've changed this now to gather up the lets and lay them out in a somewhat reasonable fashion as a kind of middle-ground between the unorthodox and orthodox methods that I think looks pretty readable.